exchange/bitswap/decision/engine.go

```diff
@@ -298,6 +298,27 @@ func (e *Engine) findOrCreate(p peer.ID) *ledger {
 	return l
 }
 
+type LedgerSnapshot struct {
+	DebtRatio     float64
+	BytesSent     uint64
+	BytesReceived uint64
+}
+
+func (e *Engine) LedgerSnapshot(p peer.ID) *LedgerSnapshot {
+	e.lock.Lock()
+	defer e.lock.Unlock()
```

Comment:
@whyrusleeping Looking at the other code I noticed you're locking on the engine. Let me know if instead I should actually lock on the ledger like you mentioned in your previous comment.